include/modem/modem_trace.h

```diff
@@ -0,0 +1,84 @@
+/*
```

Comment:
We should have the `nrf_` prefix (both for APIs and files) to align with the APIs in `nrf_modem_lib`.

Reply:
> What's the additional benefit or running these tests on qemu_cortex_m3 instead of just native_posix?

The NCS CI runs tests only on the qemu_cortex_m3 platform. So if I only have native_posix here, this test case will not be executed by the CI. I ensured that this test also executes on native_posix because it is easier to debug using gdb and also obtain code coverage reports (using twister)

Reply:
>We should have the nrf_ prefix (both for APIs and files) to align with the APIs in nrf_modem_lib.

Ok. Will work on this.

Reply:
> The NCS CI runs tests only on the qemu_cortex_m3 platform. So if I only have native_posix here, this test case will not be executed by the CI.

@balaji-nordic as I said in previous comments I suggest to leave this to platform for some reason, but for your information, testing on `native-posix` platform is now possible in CI (thanks @PerMac!). For example, if you trace logs for this PR you can observed following line:
```
22:32:36  INFO    - 76/83 native_posix              lib/modem_trace/unity.modem_trace                  PASSED (native 8.116s)
```

Reply:
>Testing on native-posix platform is now possible in CI (thanks @PerMac!)

Thats very good to hear 👍 Thanks indeed.

Reply:
@balaji-nordic Under `nrf_modem_lib` are the files that make up the modem library integration layer. Some have a public API, for example `nrf_modem_os.c` (glue) and `nrf_modem_lib.c` (wrapper), those have the `nrf_modem_` prefix both in the file and the APIs. Since this module has a public API, I would use either `nrf_modem_` or perhaps `nrf_modem_lib_` prefix to distinguish from the library's own API.

Reply:
Alright. Will do the renaming.

Reply:
@lemrey  Completed the renaming and pushed. Please have a look.